### PR TITLE
Feature: link users and roles 

### PR DESCRIPTION
**Pull request by Zorin95670**

### CODE REVIEW

File: src/test/java/com/ditrit/letomodelizerapi/service/AccessControlServiceImplTest.java [hunk 1]
```diff
@@ -129,4 +165,129 @@ void testDelete() {
...
+                .when(userAccessControlRepository.save(Mockito.any()))
+                .thenReturn(new UserAccessControl());
+
+        service.associate(AccessControlType.ROLE, 1l, "login");
+
+        Mockito.verify(userAccessControlRepository, Mockito.times(1)).save(Mockito.any());
+    }
+
+    @Test
+    @DisplayName("Test associate: should throw exception on already exists association")
```
Comment (justinehell) [inline src/test/java/com/ditrit/letomodelizerapi/service/AccessControlServiceImplTest.java:199]: `should throw exception on already existing association`

File: src/main/java/com/ditrit/letomodelizerapi/controller/RoleController.java [hunk 2]
```diff
@@ -181,4 +183,86 @@ public Response deleteRole(final @Context HttpServletRequest request,
...
+        return Response.status(this.getStatus(resources)).entity(resources).build();
+    }
+
+    /**
+     * Associates a user with a specified role.
+     * This endpoint allows an administrative user to associate another user, identified by their login,
+     * with a specific role, identified by its ID.
+     *
+     * @param request HttpServletRequest to access the HTTP session for authentication and authorization.
+     * @param id      the ID of the role with which the user is to be associated.
```
Comment (justinehell) [inline src/main/java/com/ditrit/letomodelizerapi/controller/RoleController.java:223]: the ID of the role to which the user is to be associated.

File: src/test/resources/features/Role.feature [hunk 3]
```diff
@@ -66,3 +66,74 @@ Feature: Role feature
     And  I expect response field "field" is "constraint"
     And  I expect response field "value" is "access_controls_name_type_key"
     And  I expect response field "cause" is "NOT_NULL"
+
+  Scenario: Should return all user associate to specific role
```
Comment (justinehell) [inline src/test/resources/features/Role.feature:70]: Should return all users associated with a specific role

**Review by justinehell — CHANGES_REQUESTED** [review]
`usa_id`

File: src/main/java/com/ditrit/letomodelizerapi/controller/CurrentUserController.java [hunk 4]
```diff
@@ -91,4 +104,35 @@ public Response getMyPermissions(final @Context HttpServletRequest request) {
 
         return Response.ok(permissions).build();
     }
+
+    /**
+     * Retrieves all roles associated with the current user.
```
Comment (AboubakarKhan14) [inline src/main/java/com/ditrit/letomodelizerapi/controller/CurrentUserController.java:109]: to the current user

File: src/test/java/com/ditrit/letomodelizerapi/service/AccessControlServiceImplTest.java (hunk 1 above)
Comment (Zorin95670) [reply src/test/java/com/ditrit/letomodelizerapi/service/AccessControlServiceImplTest.java:199]: done

File: src/main/java/com/ditrit/letomodelizerapi/controller/RoleController.java (hunk 2 above)
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/controller/RoleController.java:223]: done

File: src/main/java/com/ditrit/letomodelizerapi/persistence/model/UserAccessControlView.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+
+/**
+ * Represents a user access control entity in the system.
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/persistence/model/UserAccessControlView.java:14]: If it's not an entity what is it ?

File: src/main/java/com/ditrit/letomodelizerapi/controller/RoleController.java [hunk 6]
```diff
@@ -181,4 +183,86 @@ public Response deleteRole(final @Context HttpServletRequest request,
...
+     *
+     * @param request      HttpServletRequest to access the HTTP session for authentication and authorization.
+     * @param id           the ID of the role for which users are to be retrieved.
+     * @param uriInfo      UriInfo to extract query parameters for additional filtering.
+     * @param queryFilter  BeanParam object for pagination and filtering purposes.
+     * @return a Response containing a paginated list of UserDTO objects associated with the role.
+     */
+    @GET
+    @Path("/{id}/users")
+    public Response getUsersOfRole(final @Context HttpServletRequest request,
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/controller/RoleController.java:200]: done

File: src/main/java/com/ditrit/letomodelizerapi/controller/RoleController.java [hunk 7]
```diff
@@ -181,4 +183,86 @@ public Response deleteRole(final @Context HttpServletRequest request,
...
+        log.info("Received GET request to get users of role {} with the following filters: {}", id, filters);
+        Page<UserDTO> resources = accessControlService
+                .findAllUsers(id, filters, queryFilter.getPagination())
+                .map(new BeanMapper<>(UserDTO.class));
+
+        return Response.status(this.getStatus(resources)).entity(resources).build();
+    }
+
+    /**
+     * Associates a user with a specified role.
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/controller/RoleController.java:218]: done

File: src/main/java/com/ditrit/letomodelizerapi/controller/RoleController.java [hunk 8]
```diff
@@ -181,4 +183,86 @@ public Response deleteRole(final @Context HttpServletRequest request,
 
         return Response.noContent().build();
     }
+
+    /**
+     * Retrieves users associated with a specific role.
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/controller/RoleController.java:188]: done

File: src/main/java/com/ditrit/letomodelizerapi/controller/CurrentUserController.java (hunk 4 above)
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/controller/CurrentUserController.java:109]: done

**Review by justinehell — APPROVED** [review]

**Pull request merged**